src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -752,22 +845,39 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
     return null;
   }
 
-  private restartLocalVideoAfterSelection(oldStream: MediaStream, fromAcquire: boolean): void {
+  private async restartLocalVideoAfterSelection(
+    oldStream: MediaStream,
+    fromAcquire: boolean,
+    fromVideoTransformDevice: boolean
+  ): Promise<void> {
     if (
       !fromAcquire &&
       this.boundAudioVideoController &&
       this.boundAudioVideoController.videoTileController.hasStartedLocalVideoTile()
     ) {
-      this.logger.info('restarting local video to switch to new device');
-      this.boundAudioVideoController.restartLocalVideo(() => {
-        // TODO: implement MediaStreamDestroyer
-        // tracks of oldStream should be stopped when video tile is disconnected from MediaStream
-        // otherwise, camera is still being accessed and we need to stop it here.
+      if (fromVideoTransformDevice) {
+        // similar to `useWebaudio`, either Device or VideoTransformDevice, `this.activeDevices['video']` tracks the supplied inner Device.
+        // Upon in-meeting switching to VideoTransformDevice, device controller releases old "supplied" stream and
+        // calls replaceLocalVideo to avoid a full stop-start update.
+        await this.boundAudioVideoController.replaceLocalVideo();
+        this.logger.info('successfully replaced video track');
         if (oldStream && oldStream.active) {
           this.logger.warn('previous media stream is not stopped during restart video');
           this.releaseMediaStream(oldStream);
         }
-      });
+      } else {
+        // not from VideoTransformDevice, usual behavior.
+        this.logger.info('restarting local video to switch to new device');
+        this.boundAudioVideoController.restartLocalVideo(() => {
```

Comment:
Why cant we also use replaceLocalVideo here?

Reply:
the main reason to contain the blast radius.